app/models/user.rb

```diff
@@ -7,4 +7,26 @@ class User < ApplicationRecord
 
   validates :fb_id, uniqueness: true, presence: true #this can be removed when some other signup/login method is added. 
   validates :email, uniqueness: true
+
+  def badges 
+    donated_causes.uniq
+  end
+
+  def donated_causes
+    @donated_causes ||= Organization.joins(donations: :user).where("user_id = ?", id).pluck(:org_type)
+  end
+
+  ## NOTE: not yet stable. still experimenting. Need additional details. 
+  def donations_by_causes
+    return @donations_by_causes if @donations_by_causes.present?
+
+    @donations_by_causes = donated_causes.group_by(&:itself)
+                .transform_values{ |v| (v.size.to_f * 100 / donated_causes.size).round }
+                .sort_by{ |d_by_c| - d_by_c[1] }
+
+    if @donations_by_causes.size > 4 
+      @donations_by_causes = @donations_by_causes[0..2] << [ "others", @donations_by_causes[3..-1].map(&:second).reduce(:+) ]
+    end
+    @donations_by_causes
+  end
```

Comment:
So one thing I had practiced in the past in rails is not to pollute models. In similar ways of the concept of keeping controllers thin, we should try to keep models thin as well.

One way to achieve this in rails is using service objects

here is a guide https://www.toptal.com/ruby-on-rails/rails-service-objects-tutorial

What you can do is 
- Implement the change in this PR
- Create a new story on trello defining what needs to be done to refactor this.

Reply:
I don't know if a service object is really needed here. Maybe just model concerns? either way, I'm not too concerned about that for this pr because that method will probably change in some other pr that's specifically focused on the donations_by_causes section. This was just a quick way to get something showing on the page.